sourcefiles/elementrando.py

```diff
@@ -0,0 +1,550 @@
+        El.LIGHTNING: {'name': 'LightTackle'}
+    },
+    T.RED_PIN: {
+        El.SHADOW: {'name': 'Dark Pin'},
+        El.ICE: {'name': 'Ice Pin'},
+        El.LIGHTNING: {'name': 'Light Pin'}
+    },
+    T.LINE_BOMB: {
+        El.SHADOW: {'name': 'Dark Line'},
+        El.ICE: {'name': 'Ice Line'},
+        El.LIGHTNING: {'name': 'Shock Line'}
+    },
+    T.FROG_FLARE: {
+        El.SHADOW: {'name': 'FrogMatter'},
+        El.ICE: {'name': 'Frog Mist'},
+        El.LIGHTNING: {'name': 'Froginaire'}
+    },
+    T.FLAME_KICK: {
+        El.SHADOW: {'name': 'Shadow Kick'},
+        El.ICE: {'name': 'Ice Kick'},
+        El.LIGHTNING: {'name': 'Light Kick'}
+    },
+    T.BLAZE_TWISTER: {
+        El.SHADOW: {'name': 'Dark Whirl'},
+        El.ICE: {'name': 'FreezeWhirl'},
+        El.LIGHTNING: {'name': 'Shock Whirl'}
+    },
+    T.BLAZE_KICK: {
+        El.SHADOW: {'name': 'Dark Kick'},
+        El.ICE: {'name': 'FreezeKick'},
+        El.LIGHTNING: {'name': 'Shock Kick'}
+    },
+    T.FIRE_ZONE: {
+        El.SHADOW: {'name': 'Dark Zone'},
+        El.ICE: {'name': 'Ice Zone'},
+        El.LIGHTNING: {'name': 'Shock Zone'}
+    },
+    T.SWORD_STREAM: {
+        El.SHADOW: {'name': 'Dark Sword'},
+        El.FIRE: {'name': 'Sword Flame'},
+        El.LIGHTNING: {'name': 'Sword Bolt'}
+    },
+    T.ROCKET_ROLL: {
+        El.ICE: {'name': 'Ice Roll'},
+        El.FIRE: {'name': 'Flame Roll'},
+        El.LIGHTNING: {'name': 'Light Roll'}
+    },
+    T.TWISTER: {
+        El.ICE: {'name': 'IceTwist'},
+        El.FIRE: {'name': 'FlameTwist'},
+        El.LIGHTNING: {'name': 'LightTwist'}
+    },
+    T.SUPER_VOLT: {
+        El.SHADOW: {'name': 'ShadowVolt'},
+        El.ICE: {'name': 'Ice Volt'},
+        El.FIRE: {'name': 'Fire Volt'}
+    },
+    T.ANTIPODE: {
+        El.ICE: {'name': 'Icepode'},
+        El.FIRE: {'name': 'Firepode'},
+        El.LIGHTNING: {'name': 'Lightpode'}
+    },
+    T.ANTIPODE_2: {
+        El.ICE: {'name': 'Icepode 2'},
+        El.FIRE: {'name': 'Firepode 2'},
+        El.LIGHTNING: {'name': 'Lightpode 2'}
+    },
+    T.ANTIPODE_3: {
+        El.ICE: {'name': 'Icepode 3'},
+        El.FIRE: {'name': 'Firepode 3'},
+        El.LIGHTNING: {'name': 'Lightpode 3'}
+    },
+    T.ICE_WATER: {
+        El.SHADOW: {'name': 'Shadow Dark'},
+        El.FIRE: {'name': 'Flame Blaze'},
+        El.LIGHTNING: {'name': 'Light Shock'}
+    },
+    T.GLACIER: {
+        El.SHADOW: {'name': 'Dark Shadow'},
+        El.FIRE: {'name': 'Blaze Flame'},
+        El.LIGHTNING: {'name': 'Shock Light'}
+    },
+    T.FINAL_KICK: {
+        El.ICE: {'name': 'GlcierKick'},
+        El.FIRE: {'name': 'InfrnoKick'},
+        El.LIGHTNING: {'name': 'ThnderKick'}
+    },
+    T.GATLING_KICK: {
+        El.ICE: {'name': 'GlcierKick'},
+        El.FIRE: {'name': 'InfrnoKick'},
+        El.LIGHTNING: {'name': 'ThnderKick'}
+    }
+}
+
+def dual_elem(first: El, second: El, clear_enabled=False) -> El:
```

Comment:
This `clear_enabled` flag is in anticipation of another little flag we'll probably add if we think it seems decent, which is to make Shadow + Lightning become nonelemental, the way that we have Lightning + Water be Lightning and any other heterogeneous combination be Shadow currently. There's not very many techs that are actually eligible for it, so having an occasional seed with a nonelemental Omega Flare or something can be a lot of fun, and it's not random the way gear is -- you can figure out the various elements relatively easily & quickly overall.